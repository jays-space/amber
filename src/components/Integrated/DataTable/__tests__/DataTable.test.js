import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { DataTable } from "..";
import { BrowserRouter } from "react-router-dom";

const data = {
  gender: "male",
  name: {
    title: "ms",
    first: "jane",
    last: "doe",
  },
  location: {
    street: {
      number: 123,
      name: "sicamore street",
    },
    city: "johannesburg",
    state: "gauteng",
    country: "south africa",
    postcode: 1111,
    coordinates: {
      latitude: "-54.4721",
      longitude: "30.6940",
    },
    timezone: {
      offset: "+2:00",
      description: "cape town, harare",
    },
  },
  email: "jane.doe@example.com",
  login: {
    uuid: "d5ac5d23-f1f2-41c7-87fe-6d5d4eaa448c",
    username: "tinykoala839",
    password: "wp2003wp",
    salt: "f9keVBXE",
    md5: "e30841088e1974b003980f11ec19b835",
    sha1: "25e4069400001e76738ee6bf13bd524a6df2f2d8",
    sha256: "f38eff8fdee487e1888fb93dd91fd2cafecd83c9fcbe49658d1a070125854d9a",
  },
  dob: {
    date: "1989-04-24T06:25:40.777Z",
    age: 33,
  },
  registered: {
    date: "2016-09-28T11:15:47.538Z",
    age: 5,
  },
  phone: "078 225 75 50",
  cell: "075 437 25 80",
  id: {
    name: "AVS",
    value: "756.9989.7526.17",
  },
  picture: {
    large: "https://randomuser.me/api/portraits/women/16.jpg",
    medium: "https://randomuser.me/api/portraits/med/women/16.jpg",
    thumbnail: "https://randomuser.me/api/portraits/thumb/women/16.jpg",
  },
  nat: "CH",
  stage: 3,
  position: { role: "fullstack developer", team: "dev team" },
};

const mockData = [data];
const TableWithMockedRouter = ({ data }) => {
  return (
    <BrowserRouter>
      <DataTable data={data} />
    </BrowserRouter>
  );
};

describe("<DataTable />", () => {
  it("should render correctly", () => {
    render(<TableWithMockedRouter data={mockData} />);

    const dataTable = screen.getByTitle("candidates-table");
    expect(dataTable).toBeInTheDocument();
    expect(dataTable).toMatchSnapshot();
  });

  it("should render table columns", () => {
    render(<TableWithMockedRouter data={mockData} />);

    const tableColumns = screen.getAllByTestId("dataTable-column-item");
    expect(tableColumns.length).toBe(7);
  });
});
