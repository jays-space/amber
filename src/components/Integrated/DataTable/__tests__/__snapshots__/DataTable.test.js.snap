// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> should render correctly 1`] = `
<div
  class="mt-10"
  title="candidates-table"
>
  <div
    class="hidden lg:grid grid-cols-7 py-3 px-4 rounded-lg shadow-lg bg-white"
  >
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        candidate name
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        location
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        stages
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        position
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        application date
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        owner
      </p>
    </div>
    <div
      class="col-span-1"
      data-testid="dataTable-column-item"
    >
      <p
        class="tracking-wide  text-slate-700 text-sm font-bold !tracking-wide capitalize"
        data-testid="paragraph"
      >
        actions
      </p>
    </div>
  </div>
  <div
    class="my-6 rounded-lg shadow-lg bg-white"
    data-testid="table-row"
  >
    <div
      class="grid grid-rows-4 md:grid-rows-1 gap-y-10 md:grid-cols-2 lg:grid-cols-7 py-4 px-4 border-b last-of-type:border-none lg:hover:bg-slate-100 transform ease-out duration-100"
    >
      <div
        class="flex items-center"
        title="user-details"
      >
        <img
          alt="jane doe"
          class="w-16 lg:w-10 mr-4 aspect-square rounded-full false"
          src="https://randomuser.me/api/portraits/thumb/women/16.jpg"
        />
        <div
          class="flex flex-col overflow-hidden"
        >
          <p
            class="tracking-wide  text-cyan-900 font-semibold pr-4 text-sm md:text-lg capitalize whitespace-nowrap text-ellipsis overflow-hidden"
            data-testid="paragraph"
          >
            jane doe
          </p>
        </div>
      </div>
      <div
        title="candidate-city-country"
      >
        <span>
          <p
            class="tracking-wide  text-cyan-900 font-semibold w-10/12 text-lg whitespace-nowrap text-ellipsis overflow-hidden capitalize"
            data-testid="paragraph"
          >
            johannesburg
          </p>
        </span>
        <span>
          <p
            class="tracking-wide  text-slate-700 capitalize opacity-75"
            data-testid="paragraph"
          >
            south africa
          </p>
        </span>
      </div>
      <div
        class="relative z-20"
        title="application-stage"
      >
        <select
          class="w-full lg:w-fit text-lg font-semibold text-slate-700 capitalize after:absolute after:top-3 after:right-3 after:w-0 after:h-0 after:border-none focus-visible:outline-none"
          data-testid="stage-selector"
          name="stage"
        >
          <option
            class="p-10 text-lg"
            data-testid="stage-option"
            value="1"
          >
            new applied
          </option>
          <option
            class="p-10 text-lg"
            data-testid="stage-option"
            value="2"
          >
            screening
          </option>
          <option
            class="p-10 text-lg"
            data-testid="stage-option"
            value="3"
          >
            code challenge
          </option>
          <option
            class="p-10 text-lg"
            data-testid="stage-option"
            value="4"
          >
            interview
          </option>
          <option
            class="p-10 text-lg"
            data-testid="stage-option"
            value="5"
          >
            hire
          </option>
        </select>
        <div
          class=" mt-2 flex"
        >
          <div
            class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
          >
            <p
              class="tracking-wide  text-slate-700 text-white"
              data-testid="paragraph"
            >
              1
            </p>
          </div>
          <div
            class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
          >
            <p
              class="tracking-wide  text-slate-700 text-white"
              data-testid="paragraph"
            >
              2
            </p>
          </div>
          <div
            class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
          >
            <p
              class="tracking-wide  text-slate-700 text-white"
              data-testid="paragraph"
            >
              3
            </p>
          </div>
          <div
            class="px-3 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-neutral-300"
          />
          <div
            class="px-3 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-neutral-300"
          />
        </div>
      </div>
      <div
        title="position-applied"
      >
        <span>
          <p
            class="tracking-wide  text-cyan-900 font-semibold w-10/12 text-lg capitalize whitespace-nowrap text-ellipsis overflow-hidden"
            data-testid="paragraph"
          >
            fullstack developer
          </p>
        </span>
        <span>
          <p
            class="tracking-wide  text-slate-700 capitalize opacity-75"
            data-testid="paragraph"
          >
            dev team
          </p>
        </span>
      </div>
      <p
        class="tracking-wide  text-cyan-900 font-semibold text-lg"
        data-testid="paragraph"
      >
        Wed, 28 Sept 2016
      </p>
      <div
        class="flex items-center"
        title="user-details"
      >
        <img
          alt="duduzile hlongwane"
          class="w-16 lg:w-10 mr-4 aspect-square rounded-full hidden lg:flex"
          src="https://media-exp1.licdn.com/dms/image/C4D03AQELG5uq0_LIug/profile-displayphoto-shrink_100_100/0/1637782454380?e=1667433600&v=beta&t=976dwrMZyY3VL1L48bQwAOvR730al8Ad2-vGvz4pgE8"
        />
        <div
          class="flex flex-col overflow-hidden"
        >
          <p
            class="tracking-wide  text-slate-700 flex lg:hidden capitalize"
            data-testid="paragraph"
          >
            owner
          </p>
          <p
            class="tracking-wide  text-cyan-900 font-semibold pr-4 text-sm md:text-lg capitalize whitespace-nowrap text-ellipsis overflow-hidden"
            data-testid="paragraph"
          >
            duduzile hlongwane
          </p>
        </div>
      </div>
      <div
        class="hidden w-full lg:flex"
      >
        <button
          class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 bg-yellow-400 active:bg-yellow-600"
          data-testid="button-primary"
        >
          See more
        </button>
      </div>
      <div
        class="lg:hidden"
      >
        <button
          class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 bg-yellow-400 active:bg-yellow-600"
          data-testid="button-primary"
        >
          see more
        </button>
      </div>
    </div>
  </div>
</div>
`;
